### PR TITLE
Object 504 -  Remote SIM Provisioning

### PR DESCRIPTION
**Pull request by asoloway64**
new pull request template

### CODE REVIEW

File: 504.xml [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+				<Units></Units>
+				<Description><![CDATA[Profile Package is a personalised Profile using an interoperable description format that is transmitted to an eUICC to load and install a Profile.]]></Description>
+			</Item>
+			<Item ID="4">
+				<Name>Profile URI</Name>
+				<Operations>RW</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Mandatory</Mandatory>
+				<Type>String</Type>
+				<RangeEnumeration>0-255 Bytes</RangeEnumeration>
```
Comment (seanmcilroy29) [inline 504.xml]: RangeEnumeration updated to 0..255 Bytes

File: 504.xml [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+				<Units></Units>
+				<Description><![CDATA[Updates profile by using the profile package stored in Package, or, by using the profile downloaded from the Package URI. This Resource is only executable when the value of the State Resource is Downloaded.]]></Description>
+			</Item>
+			<Item ID="6">
+				<Name>State</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Mandatory</Mandatory>
+				<Type>Integer</Type>
+				<RangeEnumeration>0-3</RangeEnumeration>
```
Comment (seanmcilroy29) [inline 504.xml]: Same here RangeEnumeration updated from - to ..

File: 504.xml [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+2: Downloaded 
+3: Updating If writing the profile package to Package Resource is done, or, if the device has downloaded the profile package from the Package URI the state changes to Downloaded. Writing an empty string to Package URI Resource or setting the Package Resource to NULL (â€˜\0â€™), resets the Profile Update State Machine: the State Resource value is set to Idle and the Update Result Resource value is set to 0. When in Downloaded state, and the executable Resource Update is triggered, the state changes to Updating. If the Update Resource failed, the state returns at Downloaded. If performing the Update Resource was successful, the state changes from Updating to Idle.]]></Description>
+			</Item>
+			<Item ID="7">
+				<Name>Update Result</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Mandatory</Mandatory>
+				<Type>Integer</Type>
+				<RangeEnumeration>0-9</RangeEnumeration>
```
Comment (seanmcilroy29) [inline 504.xml]: RangeEnumeration updated from - to ..

File: 504.xml [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+
+A LwM2M client indicates the failure to retrieve the Profile using the URI provided in the Package URI resource by writing the value 9 to the /x/0/7 (Update Result resource) when the URI contained a URI scheme unsupported by the client. Consequently, the LwM2M Client is unable to retrieve the Profile using the URI provided by the LwM2M Server in the Package URI when it refers to an unsupported protocol.]]></Description>
+			</Item>
+			<Item ID="8">
+				<Name>Profile Name</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>String</Type>
+				<RangeEnumeration>0-255 Bytes</RangeEnumeration>
```
Comment (seanmcilroy29) [inline 504.xml]: Same here RangeEnumeration updated from - to ..

File: 504.xml [hunk 5]
```diff
@@ -0,0 +1,362 @@
...
+				<Units></Units>
+				<Description><![CDATA[Name of the Profile Package]]></Description>
+			</Item>
+			<Item ID="9">
+				<Name>Profile Package Version</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>String</Type>
+				<RangeEnumeration>0-255 Bytes</RangeEnumeration>
```
Comment (seanmcilroy29) [inline 504.xml]: Same here RangeEnumeration updated from - to ..

File: 504.xml [hunk 6]
```diff
@@ -0,0 +1,362 @@
...
+The above license is used as a license under copyright only.  Please
+reference the OMA IPR Policy for patent licensing terms:
+https://www.omaspecworks.org/about/intellectual-property-rights/
+-->
+
+<LWM2M xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://openmobilealliance.org/tech/profiles/LWM2M-v1_1.xsd">
+<Object ObjectType="MODefinition">
+		<Name>Remote SIM Provisioning</Name>
+		<Description1><![CDATA[This is a device management object that should be used for Remote SIM Provisioning]]></Description1>
+		<ObjectID>505</ObjectID>
```
Comment (seanmcilroy29) [inline 504.xml]: ObjectID needs to be updated to 504

Comment (asoloway64) [reply 504.xml]: Duh, I missed that one.

File: 504.xml [hunk 7]
```diff
@@ -0,0 +1,362 @@
...
+reference the OMA IPR Policy for patent licensing terms:
+https://www.omaspecworks.org/about/intellectual-property-rights/
+-->
+
+<LWM2M xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://openmobilealliance.org/tech/profiles/LWM2M-v1_1.xsd">
+<Object ObjectType="MODefinition">
+		<Name>Remote SIM Provisioning</Name>
+		<Description1><![CDATA[This is a device management object that should be used for Remote SIM Provisioning]]></Description1>
+		<ObjectID>504</ObjectID>
+		<ObjectURN>urn:oma:lwm2m:ext:504</ObjectURN>
```
Comment (seanmcilroy29) [inline 504.xml]: Object urn should be urn:oma:lwm2m:oma:504

File: 504.xml [hunk 8]
```diff
@@ -0,0 +1,362 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<!-- 
+FILE INFORMATION
+OMA Permanent Document
+   File: OMA-SUP-XML_504-V1_0-YYYY-MM-DD-A
+   Path: https://www.openmobilealliance.org/release/ObjLwM2M_Remote_SIM_Provisioning
+   Type: xml
```
Comment (seanmcilroy29) [inline 504.xml]: Following metadata to be removed as it's no longer required
  Type: xml
   Date: YYYY-MMM-DD

**Comment by jpradocueva** [comment]
IPSO is now requiring that all the new Objects should contain a value in the "ObjectVersion" and "LWM2MVersion" elements.
In this Object, the "ObjectVersion" element doesn't contain value "1.0".
IPSO needs to decide if this is acceptable or not.

**Comment by asoloway64** [comment]
Corrected ObjectVersion to be 1.0.

**Comment by mkgillmore** [comment]
Ericcson is ok with merging this 2/16/2021,  Alan needs more time.

**Comment by asoloway64** [comment]
Created new PR to avoid conflicts with DDF

**Pull request closed without merging**